uni-stark/src/prover.rs

```diff
@@ -79,52 +84,99 @@ where
         alpha,
         constraint_count,
     );
-    let quotient_flat = RowMajorMatrix::new_col(quotient_values).flatten_to_base();
+
+    let quotient_flat = RowMajorMatrix::new_col(quotient_values.clone()).flatten_to_base();
     let quotient_chunks = quotient_domain.split_evals(quotient_degree, quotient_flat);
     let qc_domains = quotient_domain.split_domains(quotient_degree);
 
     let (quotient_commit, quotient_data) = info_span!("commit to quotient poly chunks")
-        .in_scope(|| pcs.commit(izip!(qc_domains, quotient_chunks).collect_vec()));
+        .in_scope(|| pcs.commit_quotient(qc_domains.clone(), quotient_chunks.clone()));
     challenger.observe(quotient_commit.clone());
 
+    // We generate random extension field values of the size of the randomized trace randomized. Since we need `R` of degree that of the extended
+    // trace -1, we can provide `R` as is to FRI, and the random polynomial will be `(R(X) - R(z)) / (X - z)`.
+    // Since we need a random polynomial defined over the extension field, we actually need to commit to `SC::CHallenge::D`
+    // random polynomials. This is similar to flattening on the base field a polynomial over the extension field.
+    // TODO: This approach is only statistically zk. To make it perfectly zk, `R` would have to truly be an extension field polynomial.
+    let opt_r_commit_data = pcs.get_opt_randomization_poly_commitment(ext_trace_domain);
+    let (opt_r_commit, opt_r_data) = if let Some((r_commit, r_data)) = opt_r_commit_data {
+        (Some(r_commit), Some(r_data))
+    } else {
+        (None, None)
+    };
+
```

Comment:
I'd mention in the comment that we only do it if zk is enabled. Can also explicitly branch on that. E.g. something like:
```suggestion
    // We generate random extension field values of the size of the randomized trace randomized. Since we need `R` of degree that of the extended
    // trace -1, we can provide `R` as is to FRI, and the random polynomial will be `(R(X) - R(z)) / (X - z)`.
    // Since we need a random polynomial defined over the extension field, we actually need to commit to `SC::CHallenge::D`
    // random polynomials. This is similar to flattening on the base field a polynomial over the extension field.
    // TODO: This approach is only statistically zk. To make it perfectly zk, `R` would have to truly be an extension field polynomial.
     let (opt_r_commit, opt_r_data) = if config.zk() {
                   let opt_r_commit_data = pcs.get_opt_randomization_poly_commitment(ext_trace_domain).unwrap().expect("...");
                   (Some(opt_r_commit_data.0), Some(opt_r_commit_data.1))
          } else {
                  (None, None)
          };
```